configure.ac

```
@@ -3763,6 +3763,28 @@ dnl The AIX_BUILDDATE is obtained from the kernel fileset - bos.mp64
 	*) ;;
 esac
 
+# check for _Complex C type
+#
+# Note that despite most compilers define __STDC_IEC_559_COMPLEX__ - almost
+# none properly support C11+ Annex G (where pure imaginary types
+# represented by _Imaginary are mandatory).  This is a bug (see e.g.
+# llvm/llvm-project#60269), so we don't rely on presence
+# of __STDC_IEC_559_COMPLEX__.
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#include <complex.h>
+int main(void)
+{
+    double _Complex z = 1 + _Complex_I*2;
+    z = z + _Complex_I*z;
+    return (creal(z) != -1) || (cimag(z) != 3);
+}]])], [ac_cv_c_complex_supported=yes],
+[ac_cv_c_complex_supported=no],
+[ac_cv_c_complex_supported=no])
+if test "$ac_cv_c_complex_supported" = "yes"; then
+    AC_DEFINE([HAVE_C_COMPLEX], [1],
+              [Defined if _Complex C type is available.])
```

```suggestion
              [Defined if _Complex C type is available (_Imaginary may not necessarily be defined).])
```

What I understood was:

- `__STDC_IEC_559_COMPLEX__` + C99 => `_Complex/_Complex_I` are mandatory but not `_Imaginary/_Imaginary_I`.
- `__STDC_IEC_559_COMPLEX__` + C11 => `_Complex/_Complex_I` and `_Imaginary/_Imaginary_I` are mandatory.

Since we only check for `_Complex` and `_Complex_I`, it might be good to say that the macro only checks the `_Complex` support but not the `_Imaginary` one.

---

> it might be good to say that the macro only checks the _Complex support

That's exactly the message.  I don't see reasons be more verbose here.

---

I was worried that `HAVE_C_COMPLEX` could have been interpreted as "complex numbers support" in the wide sense (namely, whatever complex arithmetic you assume to be) but maybe I wanted too much precision. Your argument is sound so let's forget mine.

---

I think this conversation can be marked as resolved!